src/lib/common/sol-platform-impl-zephyr.c

```diff
@@ -85,10 +85,22 @@ sol_platform_impl_set_target(const char *target)
 }
 
 int
-sol_platform_impl_get_machine_id(char id[static 33])
+sol_platform_impl_get_machine_id(char id[SOL_STATIC_ARRAY_SIZE(33)])
 {
-    SOL_WRN("Not implemented");
+#ifdef SOL_MACHINE_ID
+    static const char ID[] = SOL_MACHINE_ID;
+    const char *ptr;
+    int i = 0;
+
+    for (ptr = ID; *ptr && i < 33; ptr++) {
+        id[i++] = *ptr;
+    }
```

Comment:
On Thu, 07 Apr 2016 13:50:24 -0700, Gustavo Lima Chaves wrote:
> >  {
> > -    SOL_WRN("Not implemented");
> > +#ifdef SOL_MACHINE_ID
> > +    static const char ID[] = SOL_MACHINE_ID;
> > +    const char *ptr;
> > +    int i = 0;
> > +
> > +    for (ptr = ID; *ptr && i < 33; ptr++) {
> > +        id[i++] = *ptr;
> > +    }
> 
> It's meant to get a string, yes, as in "4d77755382fb459abdd2181de3316aaa".

Then why not just strcpy?